### ISSUE TITLE
Option equality broken

### ISSUE THREAD

**Issue by jorge-fernandes** [emmanueltouzery/prelude-ts#74]
Hi,

stumbled upon a new equality issue

`Option.of(Vector.of(12, 2674)).equals(Option.of(null))`

yields true

i'm on  ` "prelude-ts": "1.0.5"`

**Comment by emmanueltouzery** [comment]
well, that's very crappy. I'll fix it tonight my time. I see the problem...
```ts
    if (obj === null != obj2 === null) {
        return false;
    }
```

due to operator priority the bracketing was wrong...
```ts
  if ((obj === null) != (obj2 === null)) {
    return false;
  }
```

this fixes it. I'll push and release tonight.

**Comment by emmanueltouzery** [comment]
I have now released 1.0.6 on npm. Sorry for the bad bug :grimacing:
